src/compiler/compilation.jl

```diff
@@ -273,9 +273,23 @@ function compile(@nospecialize(job::CompilerJob))
         push!(ptxas_opts, "--compile-only")
     end
 
+    ptx = job.config.params.ptx
     cap = job.config.params.cap
     arch = "sm_$(cap.major)$(cap.minor)"
 
+    # validate use of parameter memory
+    argtypes = filter([KernelState, job.source.specTypes.parameters...]) do dt
+        !isghosttype(dt) && !Core.Compiler.isconstType(dt)
+    end
+    param_usage = sum(sizeof, argtypes)
+    param_limit = 4096
+    if cap >= v"7.0" && ptx >= v"8.1"
+        param_limit = 32764
+    end
+    if param_usage > param_limit
+        error("Kernel invocation uses $(Base.format_bytes(param_usage)) of parameters, exceeding the $(Base.format_bytes(param_limit)) limit.")
```

Comment:
I was really hoping we could have a way to dump the size of each parameter, which we have found useful when trying to identify which argument is the cause of the failure (or perhaps even have a it log it always when using a debug mode?(

Reply:
Sure, that's just a matter of tweaking the error message.